### ISSUE TITLE
Custom ClusteringStrategy

### ISSUE THREAD

**Issue by andysbliss** [mg6maciej/android-maps-extensions#45]
Great job with this library - it works great!

I've found a need to extend the GridClusteringStrategy and was hoping to somehow configure the library to use the subclass without modifying the library,  Perhaps through the ClusterOptions interface. 

So first, is it correct that it is not possible currently to provide a custom ClusteringStrategy through the client interface.  And second, if not then would this be a reasonable approach (through the ClusterOptions)?

The reason I've found to extend GridClusteringStrategy is programmatic showInfoWindow does not work as expected - keeps getting ignore in onShowInfoWindow in GridClusteringStrategy.

Comments welcome.

**Comment by mg6maciej** [comment]
Whenever I had to make a custom modification of the code, I simply copied the library into a project src and just made required changes. That's the easiest way to modify it. I'd probably suggest you do that.

But what you describe sounds a bit like a bug in the library. Could you make a simple example app to show the issue? Might be as a pull request with added fragment to the demo app here ([here is a simple example issue from the past](https://github.com/mg6maciej/android-maps-extensions/blob/develop/android-maps-extensions-demo/src/main/java/pl/mg6/android/maps/extensions/demo/Issue14InfoWindowNotShowingFragment.java)) or your own project, or at least a description what your code does, what you expect and what you see instead.

Glad you like the lib.
Cheers,
M.

**Comment by andysbliss** [comment]
For the moment it's easiest just to describe...

I have a view (ListView) on top of the map and in its 'onItemClick' the code gets a reference to a Marker (m) and calls m.showInfoWindow().  I would expect the info window to show (just as if the user clicked on the marker).  The call stack leading to the issue is:

```
<my code>
   m.showInfoWindow()
      DelegatingMarker.showInfoWindow()
         MarkerManger.onShowInfoWindow()
            GridClusteringStrategy.onShowInfoWindow()

```
(now referencing the GridClusteringStrategy code) 

```
   public void onShowInfoWindow(DelegatingMarker marker) {
        if (!marker.isVisible()) {   // false
            return;
        }
        ClusterMarker cluster = markers.get(marker);
        if (cluster == null) { // false
            marker.forceShowInfoWindow();
        } else if (cluster.getMarkersInternal().size() == 1) { //false
            cluster.refresh();
            marker.forceShowInfoWindow();
        }

       // I ADDED THIS (which does what I want but may not be what is wanted in all cases?)...
        else {
            marker.forceShowInfoWindow();
        }
    }


```

The marker is a normal marker and not a cluster marker but I figure with the min cluster size at 10 it may
show as a normal marker but actual be in a cluster ?

**Comment by mg6maciej** [comment]
And the marker that was clicked in the list is actually visible in the map (you probably took a reference using `getDisplayedMarkers` or something similar?)
That's definitely a bug in the library. Should not be doing `== 1` there, but instead something like `< clusteringSettings.minClusterSize()`.

Thanks. I'll fix that soon(ish).

**Comment by andysbliss** [comment]
FWIW, here's an image of the map - the list displays the marker titles in the blue box and the user can click on a list item - basically my alternative approach to spiderfication of a bunch of markers on top of each other:
[Before](http://hmdb.andysbliss.net/images/2_0/MarkerListOnMap.png)
[After selecting item in list](http://hmdb.andysbliss.net/images/2_0/MarkerListOnMapInfo.png)

Thanks a bunch.  

One suggestion for you is to create a Logo for your library so it can be added to end products info pages (like a "intel inside" thing).

**Comment by mg6maciej** [comment]
Fixed in 2.3.1.
